fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -1314,7 +1315,8 @@ public static Expression processComparisonPredicate(ComparisonPredicate comparis
         if (GlobalVariable.enableNewTypeCoercionBehavior) {
             commonType = findWiderTypeForTwo(left.getDataType(), right.getDataType(), false, false);
         } else {
-            commonType = findWiderTypeForTwoForComparison(left.getDataType(), right.getDataType(), false);
+            commonType = findWiderTypeForTwoForComparison(left.getDataType(), right.getDataType(),
+                    comparisonPredicate instanceof EqualTo);
```

Comment:
we should not change this legacy behavior, because it will change the result of sql, lead to a behavior change after upgrade from lower version